cdap-ui/app/services/dashboard/dashboardhelper.js

```diff
@@ -0,0 +1,94 @@
+angular.module(PKG.name + '.services')
+  .factory('DashboardHelper', function (MyDataSource, MyChartHelpers, MyMetricsQueryHelper) {
+    var dataSrc = new MyDataSource();
+
+    function startPolling (widget) {
+      dataSrc.poll({
+        _cdapPath: '/metrics/query',
+        method: 'POST',
+        interval: widget.settings.interval,
+        body: MyMetricsQueryHelper.constructQuery(
+          'qid',
+          MyMetricsQueryHelper.contextToTags(widget.metric.context),
+          widget.metric
+        )
+      }, function (res) {
+
+        widget.formattedData = __formatData(res, widget);
+
+        widget.pollId = res.__pollId__;
+      });
+    }
+
+
+    function stopPolling (widget) {
+      dataSrc.stopPoll(widget.pollId);
+    }
+
+
+    function startPollDashboard (dashboard) {
+      angular.forEach(dashboard.columns, function (widget) {
+        startPolling(widget);
+      });
+    }
+
+    function stopPollDashboard (dashboard) {
+      angular.forEach(dashboard.columns, function (widget) {
+        stopPolling(widget);
+      });
+    }
+
+    function fetchData (widget) {
+      dataSrc.request({
+        _cdapPath: '/metrics/query',
+        method: 'POST',
+        body: MyMetricsQueryHelper.constructQuery(
+          'qid',
+          MyMetricsQueryHelper.contextToTags(widget.metric.context),
+          widget.metric
+        )
+      })
+      .then(function (res) {
+        widget.formattedData = __formatData(res, widget);
+
+        widget.pollId = res.__pollId__;
+      });
+    }
+
+    function fetchDataDashboard (dashboard) {
+      angular.forEach(dashboard.columns, function (widget) {
+        fetchData(widget);
+      });
+    }
+
+    function __formatData (res, widget) {
```

Comment:
Could be minor but any specific reason for ```__```? We don't have this pattern as of now in cdap I guess. I understand its to indicate its private but since we didn't expose this in line 85 isn't it implicit?